addons/website/static/src/js/editor/snippets.editor.js

```diff
@@ -70,17 +70,38 @@ weSnippetEditor.Class.include({
                 size: 'medium',
                 title: _t("Google Map API Key"),
                 buttons: [
-                    {text: _t("Save"), classes: 'btn-primary', close: true, click: async () => {
-                        const newAPIKey = dialog.$('#api_key_input').val() || false;
-                        await this._rpc({
-                            model: 'website',
-                            method: 'write',
-                            args: [
-                                [websiteId],
-                                {google_maps_api_key: newAPIKey},
-                            ],
-                        });
-                        invalidated = true;
+                    {text: _t("Save"), classes: 'btn-primary', close: false, click: async (ev) => {
+                        const $apiKeyInput = dialog.$('#api_key_input');
+                        const valueAPIKey = $apiKeyInput.val() || false;
+                        if (!valueAPIKey) {
+                            $apiKeyInput.addClass('is-invalid');
+                            dialog.$('#api_key_help').text(_t('Enter an API Key'));
+                            return;
+                        }
+                        const $button = $(ev.currentTarget);
+                        $button.prop("disabled", true);
+                        try {
+                            const response = await fetch(`https://maps.googleapis.com/maps/api/staticmap?center=belgium&size=10x10&key=${valueAPIKey}`);
+                            if (response.status === 200) {
+                                await this._rpc({
+                                    model: 'website',
+                                    method: 'write',
+                                    args: [
+                                        [websiteId],
+                                        {google_maps_api_key: valueAPIKey},
+                                    ],
+                                });
+                                invalidated = true;
+                                dialog.close();
+                            } else {
+                                $apiKeyInput.addClass('is-invalid');
+                                dialog.$('#api_key_help').text(_t('Invalid API Key'));
```

Comment:
Can't we include the google message with it? even if not translatable, it could help enduser IMO
It is not always an invalid key, eg if you have one but didnt enable the correct project, you got 
![image](https://user-images.githubusercontent.com/30048408/122033943-4f772780-cdd1-11eb-8d53-c4a1a31aa1e8.png)

The key is not really invalid.

Reply:
invalid means (for me 😅): not an api key, expired, or not authorized ...
 @sbuhl  any suggestions?

Reply:
Yes but the point is to help the user as much as we can, especially if we have the exact error..
And I just gave this example as this is the one I got when I tried with my key, pretty sure there is dozen of possible different errors. We have a very detailed info given by google, we might as well show it, even if not translatable, on top of our translatable message, eg "Invalid API Key, google returned the follow error: <google error>"